Description: Avoid deprecation warning message with Python 3.12
 This message causes two tests in dask.distributed to fail.
Forwarded: not-needed
Author: Julian Gilbey <jdg@debian.org>
Last-Update: 2024-01-01

--- a/jupyter_server/_tz.py
+++ b/jupyter_server/_tz.py
@@ -5,37 +5,15 @@
 """
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
-from datetime import datetime, timedelta, tzinfo
+from datetime import datetime, timezone
 
-# constant for zero offset
-ZERO = timedelta(0)
 
+def utcfromtimestamp(timestamp):
+    return datetime.fromtimestamp(timestamp, timezone.utc)
 
-class tzUTC(tzinfo):
-    """tzinfo object for UTC (zero offset)"""
 
-    def utcoffset(self, d):
-        return ZERO
-
-    def dst(self, d):
-        return ZERO
-
-
-UTC = tzUTC()
-
-
-def utc_aware(unaware):
-    """decorator for adding UTC tzinfo to datetime's utcfoo methods"""
-
-    def utc_method(*args, **kwargs):
-        dt = unaware(*args, **kwargs)
-        return dt.replace(tzinfo=UTC)
-
-    return utc_method
-
-
-utcfromtimestamp = utc_aware(datetime.utcfromtimestamp)
-utcnow = utc_aware(datetime.utcnow)
+def utcnow():
+    return datetime.now(timezone.utc)
 
 
 def isoformat(dt):
--- a/jupyter_server/gateway/managers.py
+++ b/jupyter_server/gateway/managers.py
@@ -20,7 +20,6 @@
 from tornado.escape import json_decode, json_encode, url_escape, utf8
 from traitlets import DottedObjectName, Instance, Type, default
 
-from .._tz import UTC
 from ..services.kernels.kernelmanager import AsyncMappingKernelManager
 from ..services.sessions.sessionmanager import SessionManager
 from ..utils import ensure_async, url_path_join
@@ -386,7 +385,7 @@
         if model:  # Update activity markers
             self.last_activity = datetime.datetime.strptime(
                 model["last_activity"], "%Y-%m-%dT%H:%M:%S.%fZ"
-            ).replace(tzinfo=UTC)
+            ).replace(tzinfo=datetime.timezone.utc)
             self.execution_state = model["execution_state"]
             if isinstance(self.parent, AsyncMappingKernelManager):
                 # Update connections only if there's a mapping kernel manager parent for
